astroquery/cadc/tests/test_cadctap.py

```diff
@@ -177,7 +177,9 @@ def raise_for_status(self):
         class CapabilitiesResponse:
             def __init__(self):
                 caps_file = data_path('tap_caps.xml')
-                self.text = open(caps_file, 'r').read()
+                with open(caps_file, 'r') as infile:
+                    text = infile.read()
+                self.text = text
```

Comment:
Shorter equivalent using `pathlib.Path`:
```suggestion
                self.text = Path(caps_file).read_text()
```

Reply:
`Path.read_text()` includes the context manager, see the source at https://github.com/python/cpython/blob/4e1303b4c9022557e04e33791f5a07c0b918b3c5/Lib/pathlib.py#L1129-L1136
My suggestion is less verbose, but the end result is the same either way.

Reply:
thanks. It was a bit hit and miss as it still seems undocumented, either case it's indeed nicer and cleaner, and may even address the one remaining mysterious open file warning coming from the remote! cadc tests (I suppose some patching gets in there somehow, I don't see it when the online tests are run, but their test collection triggers it when skipping should be done instead).